### PR TITLE
Standardise newlines after module-level docstrings

### PR DESCRIPTION
**Pull request by DanielNoord**
### Description

Resolves https://github.com/psf/black/issues/1872.

This is a rebase of https://github.com/psf/black/pull/2996 and therefore most credit should go to @jpy-git.

I have excluded the refactoring that was included in the rebase attempt at https://github.com/psf/black/pull/3287. I feel the refactoring makes the PR unnecessarily hard to review and rebase. It's probably a very desired refactor (I can't comment on this since I don't know the codebase too well), but it can be taken up in a specific PR.

_I won't autoclose those PRs with this PR as I think the refactoring should probably be rebased on its own and not lost to the closed PR list._

I would appreciate a review as I'm not sure this is the right approach for the checks, it seemed to be the best compromise between `main` and the various versions of the check that @jpy-git wrote in their original PR. Happy to change it to another check if desired. I'd really like to see this change make it into the preview so happy to help wherever needed to get this to pass.

### Checklist - did you ...

- [x] Add an entry in `CHANGES.md` if necessary?
- [x] Add / update tests if necessary?
- [x] Add new / update outdated documentation?

### CODE REVIEW

File: tests/data/module_docstring_1.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+"""Single line module-level docstring should be followed by single newline."""
```
Comment (JelleZijlstra) [inline tests/data/module_docstring_1.py:1]: I don't think we'll run this test as it's not in `simple_cases/`.

Comment (DanielNoord) [reply tests/data/module_docstring_1.py]: Moved the tests to ``preview/`` to make sure they run under preview mode. They executed now with `pytest -k module_docstring`.

Also added another two tests cases of which I thought the original PR didn't really have a good test:
1. A file that already has a new line in-between the docstring and the rest of code and
2. A file that has no lines in between the docstring and the rest of code and should thus get one

**Review by JelleZijlstra — APPROVED** [review]
Thanks! Pushed a change moving the tests as I just changed the structure of the tests.

**Comment by chalecado** [comment]
how can i exclude this from being enforced on my project? its caused 100s of files to need to change

**Pull request merged**